pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -632,13 +632,36 @@ func (og *operationGenerator) resizeFileSystem(volumeToMount VolumeToMount, devi
 				return nil, nil
 			}
 
-			diskFormatter := &mount.SafeFormatAndMount{
-				Interface: mounter,
-				Exec:      og.volumePluginMgr.Host.GetExec(expandableVolumePlugin.GetPluginName()),
+			var resizeStatus bool
+			var resizeErr error
+
+			fsResizablePlugin, callExpandFS := expandableVolumePlugin.(volume.FSResizableVolumePlugin)
+			if callExpandFS {
+				// if any plugin implements FSResizableVolumePlugin interface then call its ExpandFS method
+				// before that if the volume is "kubelet expandable" - like a flex volume, then first call
+				// its ExpandVolumeDevice method before procceeding to ExpandFS
+				if volumeToMount.VolumeSpec.IsKubeletExpandable() {
```

Comment:
What happens if volume type does not support `ExpandVolumeDevice` when volume is mounted?

Reply:
@gnufied 

1. If a kubelet expandable volume (i.e. Flex) doesn't support `ExpandVolumeDevice` at all, then as you have already suggested, its dynamic provisioner storage class will set `allowVolumeExpansion: false` and this issue is addressed
2. But if it sets `allowVolumeExpansion: true` and then returns `unsupported` from `ExpandVolumeDevice` then its probably a driver issue that needs to be addressed right

Reply:
That will make it tricky, today we only care about volumes in use. If a volume is not mounted, we don't even know which node to run this from. That should be fixed with CSI.

Reply:
@gnufied @chakri-nelluri OK I will go with #1 and amend the commit.

Reply:
@chakri-nelluri but this code will only be triggered for volumes that are in use. The reconciler checks that the volumes are in use and are mounted (and the ExpandInUsePersistentVolumes feature gate is enabled) before the code flow reaches here.

So am I missing a case here?

Reply:
What I meant is - "ExpandVolumeDevice when volume is mounted?" cannot be done and the plugin has to fail.

Reply:
If we agree to make flexvolume resize online only, that is fine. But a flexvolume plugin author must know that, it only support online volume resizing.  cc @saad-ali

Reply:
Why is `IsKubeletExpandable` check require in controller side for flex-volume plugins? Can you elaborate?

Reply:
@gnufied the `IsKubeletExpandable` check is **not** required on the controller side for flex volumes. It **is** required on the kubelet side to ensure that we call `ExpandVolumeDevice` method for flex volumes first (in `resizeFileSystem` method) _before_ calling the `ExpandFS` method.